gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/GradebookPage.java

```diff
@@ -155,6 +157,11 @@ public boolean isVisible() {
 		final List<Assignment> assignments = this.businessService.getGradebookAssignments();
 		Temp.time("getGradebookAssignments", stopwatch.getTime());
 
+		if (settings.isCategoriesEnabled()) {
+			// pre-sort assignments by the categorized sort order
+			Collections.sort(assignments, new CategorizedAssignmentComparator());
```

Comment:
I wonder if this should go in the backend and be passed into the matrix? The list of assignments is already sorted in getGradebookAssignments so this will sort twice but maybe we can't avoid that and it's not a big deal.

However note that students don't have access to the GradebookInformation so it would need to be wrapped in a role check if added to the backend.

Probably a lot of work when it could just be left here. Maybe just document why it is here and we are done.